app/integrations/client/views/integrationsOutgoing.js

```diff
@@ -49,30 +49,24 @@ Template.integrationsOutgoing.onCreated(function _integrationsOutgoingOnCreated(
 		});
 	};
 
-	this.autorun(() => {
-		const id = this.data && this.data.params && this.data.params().id;
-
-		if (id) {
-			const sub = this.subscribe('integrations');
-			if (sub.ready()) {
-				let intRecord;
-
-				if (hasAllPermission('manage-outgoing-integrations')) {
-					intRecord = ChatIntegrations.findOne({ _id: id });
-				} else if (hasAllPermission('manage-own-outgoing-integrations')) {
-					intRecord = ChatIntegrations.findOne({ _id: id, '_createdBy._id': Meteor.userId() });
-				}
-				intRecord.hasScriptError = intRecord.scriptEnabled && intRecord.scriptError;
-
-				if (intRecord) {
-					this.record.set(intRecord);
-				} else {
-					toastr.error(TAPi18n.__('No_integration_found'));
-					FlowRouter.go('admin-integrations');
-				}
-			}
+	if (params && params.id) {
+		let integration;
+		const baseUrl = `integrations.getOne?integrationId=${ params.id }`;
```

Comment:
```suggestion
		const baseUrl = `integrations.get?integrationId=${ params.id }`;
```

Reply:
I like most the approach of calling `APIClient.v1.get` with an object of params.. this way you  wouldn't need to interpolate the `baseUrl` like this.. I have to heavily rewrite it, what do you think?
```javascript
		const integration = (async () => {
			const reqParams = {
				integrationId: params.id,
			};

			if (!hasAllPermission('manage-outgoing-integrations')) {
				toastr.error(TAPi18n.__('No_integration_found'));
				FlowRouter.go('admin-integrations');
				return;
			}

			if (hasAllPermission('manage-own-outgoing-integrations')) {
				reqParams.createdBy = Meteor.userId();
			}

			const { integration } = await APIClient.v1.get('integrations.getOne', reqParams);

			return integration;
		})();

		if (!integration) {
			toastr.error(TAPi18n.__('No_integration_found'));
			FlowRouter.go('admin-integrations');
			return;
		}

		integration.hasScriptError = integration.scriptEnabled && integration.scriptError;
		this.record.set(integration);